engine/rig/src/rig.cpp

```diff
@@ -1173,83 +882,15 @@ namespace dmRig
-            const dmTransform::Transform& transform = pose[bi];
-            dmTransform::Transform& out_transform = out_pose[bi];
-            out_transform = transform;
-            if (bi > 0) {
-                const dmRigDDF::Bone* bone = &bones[bi];
-                if (bone->m_InheritScale)
-                {
-                    out_transform = dmTransform::Mul(out_pose[bone->m_Parent], transform);
-                }
-                else
-                {
-                    Vector3 scale = transform.GetScale();
-                    out_transform = dmTransform::Mul(out_pose[bone->m_Parent], transform);
-                    out_transform.SetScale(scale);
-                }
-            }
-        }
-    }
-
-    static void PoseToModelSpace(const dmRigDDF::Skeleton* skeleton, const dmArray<Matrix4>& pose, dmArray<Matrix4>& out_pose)
-    {
-        const dmRigDDF::Bone* bones = skeleton->m_Bones.m_Data;
-        uint32_t bone_count = skeleton->m_Bones.m_Count;
-        for (uint32_t bi = 0; bi < bone_count; ++bi)
-        {
-            const Matrix4& transform = pose[bi];
-            Matrix4& out_transform = out_pose[bi];
-            out_transform = transform;
-            if (bi > 0) {
-                const dmRigDDF::Bone* bone = &bones[bi];
-                if (bone->m_InheritScale)
-                {
-                    out_transform = out_pose[bone->m_Parent] * transform;
-                }
-                else
-                {
-                    Vector3 scale = dmTransform::ExtractScale(out_pose[bone->m_Parent]);
-                    out_transform.setUpper3x3(Matrix3::scale(Vector3(1.0f/scale.getX(), 1.0f/scale.getY(), 1.0f/scale.getZ())) * transform.getUpper3x3());
-                    out_transform = out_pose[bone->m_Parent] * transform;
-                }
-            }
-        }
-    }
-
-    static void PoseToInfluence(const dmArray<uint32_t>& pose_idx_to_influence, const dmArray<Matrix4>& in_pose, dmArray<Matrix4>& out_pose)
-    {
-        for (uint32_t i = 0; i < pose_idx_to_influence.Size(); ++i)
-        {
-            uint32_t j = pose_idx_to_influence[i];
-            out_pose[j] = in_pose[i];
-        }
-    }
-
-    // NOTE: We have two different vertex data write functions, since we expose two different vertex formats (spine and model).
-    // This is a temporary fix until we have better support for custom vertex formats.
     static RigModelVertex* WriteVertexData(const dmRigDDF::Mesh* mesh, const float* positions, const float* normals, RigModelVertex* out_write_ptr)
     {
         uint32_t indices_count = mesh->m_PositionIndices.m_Count;
         const uint32_t* indices = mesh->m_PositionIndices.m_Data;
         const uint32_t* uv0_indices = mesh->m_Texcoord0Indices.m_Count ? mesh->m_Texcoord0Indices.m_Data : mesh->m_PositionIndices.m_Data;
         const float* uv0 = mesh->m_Texcoord0.m_Data;
 
+        // Preferably, I'd have the vertex buffer already packed after the build pipelibne /MAWE
```

Comment:
Packed as in pre-interleaved between the channels? Maybe we should add a ticket for that, I agree with this statement

Reply:
Packed, in the sense that I don't want us to assemble the VB at runtime.

We do it because of the older GLES2, where 32bit indexbuffers amay not be supported.
I think we can leave the decision for the developer:
* Either don't create meshes with more than 65536 vertices, or
* Don't support very old devices

I think we could skip a lot of clunky code, both in content pipeline, as well as at runtime.